audit_exceptions/flat_namespace_allowlist.json

```diff
@@ -16,6 +16,7 @@
   "scalapack",
   "soapysdr",
   "soci",
+  "syslog-ng",
```

Comment:
Are you sure this isn't a consequence of the Libtool bug? If it is then we should fix it instead of adding it to the allowlist.

Reply:
Can you share a few details about the bug, please?

(To the best of my knowledge, this is not due to a bug, but because syslog-ng is currently mocking symbols in a rather weird and unportable way in unit tests.)

Reply:
Usually it comes from an old `libtool.m4` which misclassifies new macOS versions as old ones. But it seems that that's not the case here, since the flag comes from https://github.com/syslog-ng/syslog-ng/commit/ba5a85e3c19f19c82bf53ef029eeb9f9beb67a52.

This seems fine for me, but note that it would be good to stop relying on `-flat_namespace` as Apple is likely to break its usage at some point.